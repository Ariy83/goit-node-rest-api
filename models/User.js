import { Schema, model } from "mongoose";
import { handleSaveError, setUpdateSettings } from "./hooks.js";
import { emailRegexp } from "../constants/regexp.js";

const userSchema = new Schema({
  password: {
    type: String,
    required: [true, "Password is required"],
  },
  email: {
    type: String,
    match: emailRegexp,
    required: [true, "Email is required"],
    unique: true,
  },
  subscription: {
    type: String,
    enum: ["starter", "pro", "business"],
    default: "starter",
  },
  verify: {
    type: Boolean,
    default: false,
    required: true,
  },
  verificationToken: {
    type: String,
  },
  token: {
    type: String,
    default: null,
  },
  avatarURL: String,
});

userSchema.post("save", handleSaveError);

userSchema.pre("findOneAndUpdate", setUpdateSettings);
userSchema.post("findOneAndUpdate", handleSaveError);

const User = model("user", userSchema);

export default User;
