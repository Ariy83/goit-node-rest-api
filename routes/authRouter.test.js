import mongoose from "mongoose";
import request from "supertest";
import app from "../app";
import User from "../models/User";

const { DB_HOST } = process.env;

describe("test /users/login", () => {
  let server = null;

  beforeAll(async () => {
    await mongoose.connect(DB_HOST);
    server = app.listen(4000);
  });

  afterAll(async () => {
    await mongoose.connection.close();
    server.close();
  });

  beforeEach(() => {});

  afterEach(() => {});

  test("test responce", async () => {
    const loginData = {
      email: "AndriiSauliak@gmail.com",
      password: "AndriiSauliak",
    };

    const { statusCode, body } = await request(app)
      .post("/users/login")
      .send(loginData);

    expect(statusCode).toBe(200);
    expect(body.token).toBeTruthy();
    expect(body.user).toMatchSnapshot({
      email: expect.any(String),
      subscription: expect.any(String),
    });

    const testUser = await User.findOne({ email: loginData.email });
    expect(testUser).not.toBeNull();
    expect(testUser.email).toBe(loginData.email);
  });
});
