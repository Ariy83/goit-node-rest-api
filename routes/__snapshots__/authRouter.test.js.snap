// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test /users/login test responce 1`] = `
{
  "email": Any<String>,
  "subscription": Any<String>,
}
`;
